examples/common/load_dds.adb

```diff
@@ -0,0 +1,277 @@
+
+with System;
+with Interfaces;
+
+with Ada.Exceptions; use Ada.Exceptions;
+with Ada.IO_Exceptions; use Ada.IO_Exceptions;
+with Ada.Text_IO; use Ada.Text_IO;
+with Ada.Unchecked_Conversion;
+
+with GL.Types;
+with GL.Low_Level.Enums;
+with GL.Objects.Buffers;
+with GL.Objects.Textures.Targets;
+with GL.Objects.Textures.With_2D_Loader;
+with GL.Pixels;
+
+with IO;
+
+--  Extracted from SOIL.adb  Direct_Load_DDS
+procedure Load_DDS (File_Name  : String;
+                        theTexture : out GL.Objects.Textures.Texture) is
```

Comment:
Why was translating this to Ada necessary? Not that I do not want it, but I want to know the reason. I need to review this very carefully because of the usage of `Unchecked_Conversion` (which I may do later).